src/controller/SetUpCodePairer.cpp

```diff
@@ -161,8 +161,35 @@ void SetUpCodePairer::OnDiscoveredDeviceOverBleError(void * appState, CHIP_ERROR
 #endif // CONFIG_NETWORK_LAYER_BLE
 
 #if CHIP_DEVICE_CONFIG_ENABLE_DNSSD
-void SetUpCodePairer::OnDiscoveredDevice(const Dnssd::DiscoveredNodeData & nodeData)
+
+bool SetUpCodePairer::NodeMatchesCurrentFilter(const Dnssd::DiscoveredNodeData & nodeData)
 {
+    switch (currentFilter.type)
+    {
+    case Dnssd::DiscoveryFilterType::kShort:
+        return (nodeData.longDiscriminator >> 8) == currentFilter.code;
```

Comment:
Suggest masking to make sure we only look at 4 bits of shifted long discriminator.

Reply:
Added in next commit.